### PR TITLE
Extend delivery server service

### PR DESCRIPTION
**Pull request by hutnikau**
related to:
https://github.com/oat-sa/extension-tao-act/pull/111
https://github.com/oat-sa/extension-tao-delivery/pull/128

### CODE REVIEW

File: model/DeliveryServerService.php [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                $settings[TAO_DELIVERY_END_PROP] = $rEventEndDate->getTimestamp();
+            } else {
+                foreach ($rEvents as $numberOfRepetition => $rEvent) {
+                    $rEventStartDate = $rEvent->getStart();
+                    $rEventEndDate = clone $rEvent->getStart();
+
+                    $rEventEndDate->add($diff);
+
+                    $repeatedDeliveryExists = ($repeatedDeliveryService->getDelivery($delivery, $numberOfRepetition) !== false);
+                    if ($this->areWeInRange($rEventStartDate, $rEventEndDate) && !$repeatedDeliveryExists) {
```
Comment (antoinerobin) [inline model/DeliveryServerService.php]: If there is no delivery currently available we display the start and end time of the parent delivery, can we easily get the next available delivery start and end time ?

**Comment by antoinerobin** [comment]
When you create a delivery assigned to a group with sub deliveries and exclude one test taker from a sub delivery (or more) and then re include it, the test taker will see the parent delivery and the ones where you modify the exclusion rule. He can take either the parent one or the one that is currently available

**Comment by hutnikau** [comment]
I have changed logic of retrieving available deliveries. 
Please make sure that you checked out this branch in the taoDelivery ext:
https://github.com/oat-sa/extension-tao-delivery/pull/130

File: model/DeliveryServerService.php [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+                $properties[RepeatedDeliveryService::PROPERTY_NUMBER_OF_REPETITION] = $numberOfRepetition;
+            }
+        } else {
+            $properties[TAO_DELIVERY_START_PROP] = (!(is_object($propStartExec)) or ($propStartExec=="")) ? null : $propStartExec->literal;
+            $properties[TAO_DELIVERY_END_PROP] = (!(is_object($propEndExec)) or ($propEndExec=="")) ? null : $propEndExec->literal;
+            $properties[DeliveryScheduleService::TAO_DELIVERY_RRULE_PROP] = (!(is_object($rrule)) or ($rrule=="")) ? null : $rrule->literal;
+        }
+
+        $properties[TAO_DELIVERY_MAXEXEC_PROP] = (!(is_object($propMaxExec)) or ($propMaxExec=="")) ? 0 : $propMaxExec->literal;
+        $properties[CLASS_COMPILEDDELIVERY] = $delivery;
```
Comment (antoinerobin) [inline model/DeliveryServerService.php:300]: The delivery you get here is the parent one and you use it to generate the url to start the delivery execution. It tries to start the parent delivery even if you try to execute a repeated one. So you got a 500 error at checkTime

Comment (hutnikau) [inline model/DeliveryServerService.php:300]: I have pushed new commit, please review it again.
Please make sure that this PR merged before testing:
https://github.com/oat-sa/extension-tao-delivery/pull/131

**Pull request merged**